jupyterlab/commands.py

```diff
@@ -1475,14 +1476,17 @@ def sort_key(key_value):
         for version, data in sorted(versions.items(),
                                     key=sort_key,
                                     reverse=True):
-            # skip deprecated versions
-            if 'deprecated' in data:
-                continue
-
             deps = data.get('dependencies', {})
             errors = _validate_compatibility(name, deps, core_data)
             if not errors:
                 # Found a compatible version
+                # skip deprecated versions
```

Comment:
Should we also do this in the other place we skip deprecated packages? I'm not sure when that is being used vs this one.

Reply:
I hid a little note in the commit message:
> Only log in `_latest_compatible_package_version`, not `latest_compatible_package_versions`, due to the difference in use: `_latest_compatible_package_version` is only called from install/update commands.

Reply:
The other version is typically called by the extension manager, which is trying to check if any of the installed extensions are outdated (if so, display an "Update" button).